src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -159,13 +225,74 @@ private XmlNode FindIgnorables(XmlDocument document)
 					.OfType<XmlNode>()
 					.Where(e => e.Prefix == nspace.Name);
 
-				if (excludeNodes.Any() && ignorables == null)
+				if (ignorables == null && excludeNodes.Any())
 				{
-					throw new Exception($"Declare \"mc:Ignorable=\"d\" in namespace of {document.BaseURI} to ignore xaml for " + excludeNamespaces.Select(n => n.Name).JoinBy(", "));
+					shouldCreateIgnorable = true;
 				}
 			}
 
-			return ignorables;
+			return (ignorables, shouldCreateIgnorable);
+		}
+
+		/// <summary>
+		/// Returns those XAML namespace definitions for which a conditional is set, grouped by those for which the conditional returns true and
+		/// should be included, and those for which it returns fales and should be excluded.
+		/// </summary>
+		private (List<XmlAttribute> IncludedConditionals, List<XmlAttribute> ExcludedConditionals) FindConditionals(XmlDocument document)
+		{
+			var included = new List<XmlAttribute>();
+			var excluded = new List<XmlAttribute>();
+
+			foreach (XmlAttribute attr in document.DocumentElement.Attributes)
+			{
+				if (attr.Prefix != "xmlns")
+				{
+					// Not a namespace
+					continue;
+				}
+
+				var valueSplit = attr.Value.Split('?');
+				if (valueSplit.Length != 2)
+				{
+					// Not a (valid) conditional
+					continue;
+				}
+
+				if (ShouldInclude() is bool shouldInclude)
+				{
+					if (shouldInclude)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'; some subsequent code is never executed.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)